docs/features/shared-volumes-with-NFS.md

```diff
@@ -0,0 +1,129 @@
+# Supporting NFS shared volumes with vDVS
+
+## Requirements
+* Developer should be able to use externally managed NFS share as “shared docker volume” using “vsphere” volume driver.
+* This allows them to use NFS directories as docker volumes using Docker APIs, Compose etc 
+* Extend the vSphere Docker Volume plugin to support NFS shared volumes and multiple readers/writers access
+* Volumes are accessible on multiple docker hosts running on  multiple ESX hosts in a Docker swarm
+* Allow admin to specify what remote dirs are managed by the plugin
+* Support all current docker volume plugin API for shared volumes
+
+## User experience
+### Set up remote directories for docker volumes
+The design here assumes that the admin is responsible to identify remote directories to export and has configured remote servers to export those. Specifically, the admin needs to have,
+1. Exported remote directories to be used to host docker volumes.
+1. Configured sufficient permissions for the admins on the Docker host to be able to mount and create folders and files on those directories.
+1. The docker host has the necessary packages of the desired version of NFS client installed.
+
+The admin can use the same remote directory configuration on multiple docker hosts on separate physical hosts and be able to view and use the same volumes on their containers (the applications must be able to synchronize access to shared volumes).
+
+### Update plugin configuration
+The plugin needs to be provided the details of the exported diretories and the servers from which to mount them. The details are entered into the plugin configuration file at /etc/docker-volume-vsphere.conf in the format as detailed later.
+
+### Restart plugin
+The plugin needs to be restarted to load the updated configuration.
+
+### Using shared volumes
+Docker volume commands must be used as shown below to use volumes on remote directories. Similar to how local VMDK based volumes are qualified with a datastore name, shared volumes must also be qualified with the label (see remote directory configuration below) that identifies the remote dir. to use to locate and refer a volume.
+1. docker volume create - 
+   1. _docker volume create --driver=vsphere --name=remote-vol -o type=<fstype - nfs/nfs4> - uses the default entry in the list of remote dirs to locate the volume
```

Comment:
I have the same concern actually. Ideally having just one driver name will make it easier for the user to use our product. However, my concern is it might be difficult to handle all use cases in a consistent way between these two modes, such as volume inspection, multi-tenancy, etc. If we can't handle all those cases consistently, it will be awkward to have the same driver name for two distinct usage modes.